packages/backend/src/database/migrations/20240314104022_migrate_space_share_table_data.ts

```diff
@@ -0,0 +1,178 @@
+                    ).andOn(
+                        `${PROJECTS_TABLE}.project_uuid`,
+                        '=',
+                        `${PROJECT_GROUP_ACCESS_TABLE}.project_uuid`,
+                    );
+                },
+            )
+            .groupBy(
+                `${USER_TABLE}.user_uuid`,
+                `${SPACE_TABEL}.space_uuid`,
+                `${PROJECT_MEMBERSHIPS_TABLE}.role`,
+                `${ORGANIZATION_MEMBERSHIPS_TABLE}.role`,
+            )
+            .select<
+                {
+                    user_uuid: string;
+                    space_uuid: string;
+                    project_role: ProjectMemberRole | null;
+                    organization_role: OrganizationMemberRole;
+                    group_roles: (ProjectMemberRole | null)[];
+                }[]
+            >([
+                `${USER_TABLE}.user_uuid`,
+                `${SPACE_TABEL}.space_uuid`,
+                `${PROJECT_MEMBERSHIPS_TABLE}.role as project_role`,
+                `${ORGANIZATION_MEMBERSHIPS_TABLE}.role as organization_role`,
+                knex.raw(
+                    `array_agg(${PROJECT_GROUP_ACCESS_TABLE}.role) as group_roles`,
+                ),
+            ]);
+
+        const spaceAccessRows = (await rows).reduce<CreateDbSpaceUserAccess[]>(
+            (
+                acc,
+                {
+                    user_uuid,
+                    space_uuid,
+                    project_role,
+                    organization_role,
+                    group_roles,
+                },
+            ) => {
+                const inheritedOrgRole: OrganizationRole = {
+                    type: 'organization',
+                    role: convertOrganizationRoleToProjectRole(
+                        organization_role,
+                    ),
+                };
+
+                const inheritedProjectRole: ProjectRole = {
+                    type: 'project',
+                    role: project_role ?? undefined,
+                };
+
+                const inheritedGroupRoles: GroupRole[] = group_roles.map(
+                    (role) => ({ type: 'group', role: role ?? undefined }),
+                );
+
+                const highestRole = getHighestProjectRole([
+                    inheritedOrgRole,
+                    inheritedProjectRole,
+                    ...inheritedGroupRoles,
+                ]);
+
+                if (!highestRole) {
+                    return acc;
+                }
+
+                return [
+                    ...acc,
+                    {
+                        user_uuid,
+                        space_uuid,
+                        space_role: convertProjectRoleToSpaceRole(
+                            highestRole.role,
+                        ),
+                    },
+                ];
+            },
+            [],
+        );
+
+        await knex(SPACE_USER_ACCESS).insert(spaceAccessRows);
```

Comment:
After inserting we should delete `space_share`. On `down()` we need to recreate it and migrate the data.